client/src/components/Workflow/Editor/Forms/FormTool.vue

```diff
@@ -0,0 +1,177 @@
+                textEnable="Set in Advance"
+                textDisable="Set at Runtime"
+                @onChange="onChange"
+                ref="form"
+            />
+            <FormSection :id="id" :getNode="getNode" :datatypes="datatypes" @onChange="onChangeSection" />
+        </template>
+    </FormCardTool>
+</template>
+
+<script>
+import axios from "axios";
+import { getAppRoot } from "onload/loadConfig";
+import Form from "components/Form/Form";
+import FormCardTool from "components/Form/FormCardTool";
+import FormSection from "./FormSection";
+import FormElement from "components/Form/FormElement";
+import { checkLabels } from "components/Workflow/Editor/modules/utilities";
+import Utils from "utils/utils";
+
+export default {
+    components: {
+        Form,
+        FormCardTool,
+        FormElement,
+        FormSection,
+    },
+    props: {
+        datatypes: {
+            type: Array,
+            required: true,
+        },
+        getManager: {
+            type: Function,
+            required: true,
+        },
+        getNode: {
+            type: Function,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            errorLabel: null,
+            mainValues: {},
+            sectionValues: {},
+            messageVariant: "",
+            messageText: "",
+        };
+    },
+    computed: {
+        node() {
+            return this.getNode();
+        },
+        workflow() {
+            return this.getManager();
+        },
+        id() {
+            return `${this.node.id}:${this.node.config_form.id}`;
+        },
+        inputs() {
+            const inputs = this.node.config_form.inputs;
+            Utils.deepeach(inputs, (input) => {
+                if (input.type) {
+                    if (["data", "data_collection"].indexOf(input.type) != -1) {
+                        input.hiddenInWorkflow = true;
+                        input.info = `Data input '${input.name}' (${Utils.textify(input.extensions)})`;
+                        input.value = { __class__: "RuntimeValue" };
+                    } else if (!input.fixed) {
+                        input.connectable = true;
+                        input.collapsible_value = {
+                            __class__: "RuntimeValue",
+                        };
+                        input.is_workflow =
+                            (input.options && input.options.length === 0) ||
+                            ["integer", "float"].indexOf(input.type) != -1;
+                    }
+                }
+            });
+            Utils.deepeach(inputs, (input) => {
+                if (input.type === "conditional") {
+                    input.connectable = false;
+                    input.test_param.collapsible_value = undefined;
+                }
+            });
+            return inputs;
+        },
+    },
+    methods: {
+        onAnnotation(newAnnotation) {
+            this.node.setAnnotation(newAnnotation);
```

Comment:
This and all the other functions that call methods on node should be events, right ?

Reply:
Turned into an event.